cli/src/lib/chain/offer/offer.ts

```diff
@@ -315,68 +291,76 @@ ${await offersInfoToString([offerInfoErrors, offersInfo])}
 `);
 }
 
-async function addCPs({
-  CPs,
-  offerId,
-  market,
-  offerName,
-}: {
-  CPs: CPFromProviderConfig[];
+type AddRemainingCPsOrCUsArgs = {
+  allCPs: CPFromProviderConfig[];
+  addedCPs: CPFromProviderConfig[];
   offerId: string;
   market: IMarket;
   offerName: string;
+};
+
+export async function addRemainingCPs({
+  allCPs,
+  addedCPs = [],
+  offerId,
+  market,
+  offerName,
+}: Omit<AddRemainingCPsOrCUsArgs, "addedCPs"> & {
+  addedCPs?: CPFromProviderConfig[];
 }) {
-  const CUsToRegisterCount = CPs.flatMap(({ unitIds }) => {
-    return unitIds;
-  }).length;
-
-  const { registeredValues: registeredCPs } = await guessTxSizeAndSign({
-    sliceValuesToRegister: sliceCPsByNumberOfCUs(CPs),
-    sliceIndex: CUsToRegisterCount,
-    getArgs(CPsToRegister) {
-      return [offerId, CPsToRegister];
-    },
-    getTitle({ valuesToRegister: CPsToRegister }) {
-      return `Add compute peers:\n${CPsToRegister.map(
-        ({ name, peerIdBase58 }) => {
-          return `${name} (${peerIdBase58})`;
-        },
-      ).join("\n")}\n\nto offer ${offerName} (${offerId})`;
-    },
-    method: market.addComputePeers,
-  });
+  let totalAddedCPs = addedCPs.length;
 
-  await addRemainingCUsToCPs({
-    allCPs: CPs,
-    registeredCPs,
-    offerId,
-    market,
-    offerName,
-  });
+  while (totalAddedCPs < allCPs.length) {
+    const remainingCPs = allCPs.slice(totalAddedCPs);
+
+    const CUsToRegisterCount = remainingCPs.flatMap(({ unitIds }) => {
+      return unitIds;
+    }).length;
+
+    const { registeredValues: addedCPs } = await guessTxSizeAndSign({
```

Comment:
can `addedCPs` ever be empty due to some error handling inside `guessTxSizeAndSign`?

I'm vary of this `while` becoming an infinite loop. Maybe it makes sense to exit with error if `addedCPs` is empty?